File: w4/static/Identifable/Identifiable.cpp
#include "Identifiable.h"

Identifiable::Identifiable() {
    this->uID = nextUID;
    nextUID += 1;
}

int Identifiable::getNextUID(){
    return nextUID;
}

int Identifiable::getUID() const {
    return this->uID;
}

int Identifiable::nextUID = 0;

File: w4/php_dz/Identifiable/Identifiable.php
<?php
    class Identifiable {
        private static $nextUID = 0;
        private $uID;

        public function Identifiable() {
            $this->uID = self::$nextUID;
            self::$nextUID += 1;
        }

        public static function getNextUID() {
            return self::$nextUID;
        }
    }
?>